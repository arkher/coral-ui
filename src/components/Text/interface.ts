import { TextProps } from '@shopify/restyle';
import { ReactNode } from 'react';
import { Theme } from '../../themes/institucional';

type CustomTextProps = Partial<{
  children?: ReactNode;
  fs?: Custom.FontSize;
  lh?: Custom.LineHeight;
  ls?: Custom.LetterSpacing;
  testID?: string;
}> &
  Partial<TextProps<Theme>>;

export { CustomTextProps };
