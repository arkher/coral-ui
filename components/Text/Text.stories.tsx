import React from "react";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react-native";
import Text from "./Text";

export default {
  title: "Text",
  component: Text,
};

// Stories
export const Basic = () => <Text>What's up</Text>;

// Add all stories to RN/Expo storybook
storiesOf("Text", module).add("Basic", Basic);
