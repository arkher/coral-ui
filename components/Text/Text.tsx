import { createText } from "@shopify/restyle";
import { Theme } from "../../themes/default";

const Text = createText<Theme>();

export default Text;
