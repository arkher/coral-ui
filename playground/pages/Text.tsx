import React from 'react';
import { Box, Text } from '../../src/components';

const Playground: React.FC = () => (
  <Box
    padding="sm"
    flexDirection="column"
    justifyContent="space-between"
    flex={1}
  >
    <Box>
      <Text fs="md" pb="nano">
        <Text fontWeight="bold">Text elements</Text> - ðŸ” 
      </Text>
    </Box>
  </Box>
);

export default Playground;
