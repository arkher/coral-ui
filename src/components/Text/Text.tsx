import React from 'react';
import { createRestyleComponent, createText } from '@shopify/restyle';
import {
  fontSize,
  letterSpacing,
  lineHeight,
  Theme,
} from '../../themes/default';
import { CustomTextProps } from './interface';

const Text = createText<Theme>();

const CustomText: React.FC<CustomTextProps> = ({ children, ...props }) => (
  <Text variant="regular" {...props}>
    {children}
  </Text>
);

export default createRestyleComponent<CustomTextProps, Theme>(
  [fontSize, lineHeight, letterSpacing],
  CustomText,
);
