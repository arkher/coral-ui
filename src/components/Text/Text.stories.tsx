import React from 'react';
import { storiesOf } from '@storybook/react-native';
import Text from './Text';
// import { fontSizes, letterSpacings, lineHeights } from '../../themes/default';

export default {
  title: 'Elements/Text',
  component: Text,
};

// Stories
// eslint-disable-next-line arrow-body-style
export const BasicText = (): React.ReactNode => {
  return (
    <Text m="sm">
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Reiciendis rerum
      esse modi natus laborum dolorum quidem omnis alias autem! Beatae ut quo
      magnam rerum soluta labore laudantium? Quasi, molestias et!
    </Text>
  );
};

// Add all stories to RN/Expo storybook
storiesOf('Text', module).add('Basic', BasicText);
