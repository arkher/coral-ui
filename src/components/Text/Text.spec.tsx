import React from 'react';
import { cleanup, render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import themeMaestro from '../../themes/maestro';
import Text from './Text';

afterEach(cleanup);

test('should Text render correctly', async () => {
  const { toJSON } = render(
    <ThemeProvider theme={themeMaestro}>
      <Text>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. In totam
        dolorum ab vel ratione blanditiis amet tempora et doloribus dignissimos
        iste quia rerum, laudantium nesciunt. Expedita non minus quis culpa!
      </Text>
    </ThemeProvider>,
  );

  expect(toJSON()).toMatchSnapshot();
});
