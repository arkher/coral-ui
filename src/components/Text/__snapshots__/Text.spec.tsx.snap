// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Text render correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "display": "flex",
        "fontFamily": "\\"Inter\\", sans-serif",
        "fontWeight": "400",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. In totam dolorum ab vel ratione blanditiis amet tempora et doloribus dignissimos iste quia rerum, laudantium nesciunt. Expedita non minus quis culpa!
</Text>
`;
